### ISSUE TITLE
Implement messaging in PBEM games

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#477]
It seems there was supposed to be a messaging system in Modem and PBEM games: news.cpp checks for it and has the newscast tell you to check the Tracking Station for a message from your opponent.  It would be handy if we implemented that at some point, probably after our 2.0 release.

The messaging system itself doesn't seem to have ever been written: I started a PBEM game in the CD version, and nothing happens when you click on the Tracking Station (which I assume is also where you would go to send a message): or rather, you just see the help message saying "One of the most important functions of a space launch complex is its tracking facilities."

**Comment by peyre** [comment]
I've found another artifact of this feature - there's a help item for it, I811:

THE TRACKING STATION IS WHERE YOU CAN READ YOUR OPPONENT'S LAST INCOMING MESSAGE, AND WRITE AN OUTGOING MESSAGE THAT WILL BE SENT AT THE END OF THE NEXT TURN.  THE NOTIFICATION OF A NEW MESSAGE WILL BE IN THE NEWS.

**Comment by pemensik** [comment]
It is not very good idea to implement messaging. If you do that, it may make the game unsuitable for children. Not likely <13 year children would want to play with it anyway. But messaging with people puts limits to PEGI and similar scores. Because there are clearly far superior communicators these days, I am not sure this is needed.

**Comment by peyre** [comment]
Well, games like Master of Orion II and Birth of the Federation had messenging systems, not to mention MMORPGs nowadays.  I don't really see the problem.

**Comment by rodneyrod** [comment]
I guess if the host (initiator?) has an option to disable messaging then that shouldn't be an issue. Not that it's an issue right now since it's not implemented at the moment.

**Comment by hweimer** [comment]
I think the bigger question is how two players are supposed to get connected in the first place. If we rely entirely on outside communications channels that would still not solve the problem of initial contact. Additionally, people might be hesitant to share personal data such as e-mail addresses or IM handles because of obvious spam problems. A better solution would be to have either a centralized server or a decentralized P2P network, which could even be used to have general discussions with other players.

**Comment by peyre** [comment]
I've always assumed that if two people are gearing up to use the PBEM feature, they have a passing familiarity anyway, at least enough to share emails.  If you started a game with someone who turned out to be a creep, you don't have to load the save and keep playing (in other words I don't see it as any different from any other situation where you've shared your email with someone).

I'm not a fan of having a centralized server.  How long would we maintain it?  And then the feature wouldn't be usable.  Like today you can't play the original Neverwinter Nights MMORPG because the central server shut down.  However you can still play BARIS (CD version) by mail with someone because it doesn't rely on an centralized server.  A decentralized PtP network might be ok, though it sounds like a lot to build.

**Comment by hweimer** [comment]
I'm also not a fan of a centralized system. I was more thinking in the direction of using an existing P2P network such as Bittorrent. Sounds at least fun to implement, I will take a look at it once the serialization stuff and the related security issues are taken care of.

**Comment by peyre** [comment]
Of course, if you're looking for security and convenience, there's always the option to add TCP support (#174).

**Comment by peyre** [comment]
Found a related artifact: lines 787-90 in game_main.cpp:

`        if (MAIL == 1) { // Hand back to the U.S. side
            MailSwitchPlayer();
            return;
        }`

**Comment by peyre** [comment]
I get that this in-game messaging feature is kind of a marginal option - there are much more convenient ways to communicate now, not least being text messaging.  I suppose it's an artifact of 1990s telecommunication and was meant largely for players using the Modem option, since most people were on dial-up back then and if you were in the middle of a Modem game you couldn't check your email, with BARIS hogging the modem.

I just think it would be good to implement since it was supposed to be in the game but wasn't followed through back in the day.
